src/features/posts/models/Posts.model.ts

```diff
@@ -0,0 +1,25 @@
+export type BasePostModel = {
+  readonly id: number;
+  readonly media?: string;
+  readonly userId: number;
+  readonly content: string;
+  readonly categoryName: string;
+  readonly createdAt: Date;
+  readonly updatedAt: Date;
+};
+
+export type BasePostRequestModel = Pick<
+  BasePostModel,
+  "content" | "media" | "categoryName"
+>;
+
+export type CreatePostRequestModel = Omit<BasePostModel, "id">;
+export type DeletePostRequestModel = Pick<BasePostModel, "id" | "userId">;
+
+export type LikePostModel = Pick<BasePostModel, "id" | "userId">;
+export type DeletePostLikeModel = Pick<BasePostModel, "id" | "userId">;
+
+export type PostIdBodyProps = {
```

Comment:
I would name this `UpdatePostRequestModel` instead.
And also type `postId `as a number.